mono/mini/intrinsics.c

```diff
@@ -109,80 +109,135 @@ llvm_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 {
 	MonoInst *ins = NULL;
 	int opcode = 0;
+	// Convert Math and MathF methods into LLVM intrinsics, e.g. MathF.Sin -> @llvm.sin.f32
+	if (in_corlib && !strcmp (m_class_get_name (cmethod->klass), "MathF") && cfg->r4fp) {
+		// (float)
+		if (fsig->param_count == 1 && fsig->params [0]->type == MONO_TYPE_R4) {
+			if (!strcmp (cmethod->name, "Ceiling")) {
```

Comment:
It might be useful to add an intrinsics-methods.h file similar to simd-methods-netcore.h and do a binary search.